### ISSUE TITLE
When using blueprint, filesPath hook not receiving options parameter

### ISSUE THREAD

**Issue by IAmJulianAcosta** [ember-cli/ember-cli#8866]
Ember CLI: 3.12.0

`filesPath()` is being called in `files()` that has [caching](https://github.com/ember-cli/ember-cli/blob/master/lib/models/blueprint.js#L235-L237), so it's only getting called once.

When is called for the [first time](https://github.com/ember-cli/ember-cli/blob/master/lib/models/blueprint.js#L716), [this.options](https://github.com/ember-cli/ember-cli/blob/master/lib/models/blueprint.js#L447) doesn't have a value, so the [value is never read by filesPath()](https://github.com/ember-cli/ember-cli/blob/master/lib/models/blueprint.js#L239)

There are several options for this:

1. this.options is being available in `_getFilesForInstall()`
2. remove cache for `files()`
3. add a flag to force recompute `files()`

**Comment by rwjblue** [comment]
Can you help me understand your use case a bit better (e.g. why do you need `files()` to have access to `this.options`)?

**Comment by IAmJulianAcosta** [comment]
One possible case is when an user needs to customize the path for the blueprint files based on some parameter passed when running `ember g blueprint --param=foo`.

In fact I don't need it, I was looking for `fileMapTokens()` instead of `filesPath()`, but there might be someone that will need it

**Comment by km-smith** [comment]
I ran into this when I was trying to have a single command say `ember g page path/to/page --type=listing` where I was planning to switch files based on the `type` option. I got the idea from the [docs](https://cli.emberjs.com/release/advanced-use/blueprints/#filespath). It was a bit confusing because the hook gets called once with `options` as `undefined` then a few more times with the actual options but it only seems to care about that first time.
